lib/Sema/CSDiagnostics.cpp

```diff
@@ -6224,3 +6224,23 @@ bool CoercionAsForceCastFailure::diagnoseAsError() {
       .highlight(getSourceRange());
   return true;
 }
+
+bool KeyPathRootTypeMismatchFailure::diagnoseAsError() {
+  auto &cs = getConstraintSystem();
+  auto locator = getLocator();
+  auto anchor = locator->getAnchor();
+
+  assert(locator->isKeyPathRoot() && "Expect a key path root");
+
+  if (auto *subscriptExpr = dyn_cast_or_null<SubscriptExpr>(anchor)) {
```

Comment:
I think this should be a cast because `[keyPath:]` is always a subscript right?

Reply:
Also please use `castToExpr<SubscriptExpr>` here instead since `anchor` is not a `Expr *` anymore.

Reply:
Actually `locator->getAnchor()`  -> `getAnchor()` and `emitDiagnosticAt` should be `emitDiagnostic` because it would already use that anchor.

Reply:
Make sense, fixed :)